File: src/main/java/no/novari/flyt/egrunnerverv/gateway/dispatch/converting/InstanceFlowHeadersToInstanceHeadersEntityConvertingService.java
package no.novari.flyt.egrunnerverv.gateway.dispatch.converting;

import no.novari.flyt.egrunnerverv.gateway.dispatch.model.InstanceHeadersEntity;
import no.novari.flyt.kafka.instanceflow.headers.InstanceFlowHeaders;
import org.springframework.stereotype.Service;

@Service
public class InstanceFlowHeadersToInstanceHeadersEntityConvertingService {
    public InstanceHeadersEntity convert(InstanceFlowHeaders instanceFlowHeaders) {
        return InstanceHeadersEntity
                .builder()
                .sourceApplicationIntegrationId(instanceFlowHeaders.getSourceApplicationIntegrationId())
                .sourceApplicationInstanceId(instanceFlowHeaders.getSourceApplicationInstanceId())
                .archiveInstanceId(instanceFlowHeaders.getArchiveInstanceId())
                .build();
    }
}

File: src/main/kotlin/no/novari/flyt/egrunnerverv/gateway/dispatch/converting/InstanceFlowHeadersToInstanceHeadersEntityConvertingService.kt
package no.novari.flyt.egrunnerverv.gateway.dispatch.converting

import no.novari.flyt.egrunnerverv.gateway.dispatch.model.InstanceHeadersEntity
import no.novari.flyt.kafka.instanceflow.headers.InstanceFlowHeaders
import org.springframework.stereotype.Service

@Service
class InstanceFlowHeadersToInstanceHeadersEntityConvertingService {
    fun convert(instanceFlowHeaders: InstanceFlowHeaders): InstanceHeadersEntity =
        InstanceHeadersEntity(
            sourceApplicationIntegrationId = instanceFlowHeaders.sourceApplicationIntegrationId,
            sourceApplicationInstanceId = instanceFlowHeaders.sourceApplicationInstanceId,
            archiveInstanceId = instanceFlowHeaders.archiveInstanceId,
        )
}
